### PR TITLE
Implement new styleProperties endpoint in the frontend and fix containerId issue

### LINKED ISSUE #33699: [TASK] Create a REST API Endpoint(s) for Style Properties Management
### Description

Implement a REST API endpoint to support the insertion and updating of style properties for `Contentlets`. The API should provide a clean interface for managing styles while ensuring data consistency and validation against UVE schemas.

1. Create the REST endpoint at /v1/page/{pageId}/styles

    Implement PUT endpoint for style properties management
    Follow dotCMS REST API conventions
    Ensure proper authentication and authorization
  
2. Define and implement the request payload structure

    Include MultiTree reference attributes:
    
    child - Contentlet identifier
    parent1 - Container identifier
    parent2 - Page identifier
    relation_type - Relationship type
    personalization - Personalization identifier
    variant_id - Variant identifier
    
    Include the styles data object
    Document payload structure with swagger 

 3. Implement validation for style properties

    Validate that styles are consistent with UVE schemas
    Implement schema validation logic
    Return appropriate error messages for validation failures
  
 4. Implement business logic for inserting/updating styles

    Handle creation of new style entries
    Handle updates to existing style entries
    Ensure proper MultiTree reference handling
    Implement transactional behavior
  
5. Write comprehensive tests

    Unit tests for validation logic
    Integration tests for API endpoints
    Test various payload scenarios (valid, invalid, edge cases)
    Test UVE schema validation
    Test authentication and authorization

6. Hide this functionality under the feature flag

  Ensure endpoint is only active when FEATURE_FLAG_UVE_STYLE_EDITOR feature flag is enabled
  Return appropriate response when feature is disabled

 4.7 Update API documentation

### Acceptance Criteria

- [ ] REST endpoint /v1/page/{pageId}/styles is implemented and functional
- [ ] Payload correctly includes all MultiTree reference attributes and styles data
- [ ] Style validation against UVE schemas is working correctly
- [ ] All tests pass (unit and integration)
- [ ]  Proper error handling and meaningful error messages are returned

### Priority

None

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by dario-daza**
This PR updates the endpoint used for `styleProperties` definition in the FE. Plus a fix was added in the BE to extend the support of **containerIds**, now you can use container IDs like this:
```json
"identifier": "83baa82257cb262c122b3a6ea032ea7f"
"identifier": "/application/containers/mycontainer/"
```

### Proposed Changes
* `dot-page-api.service.spec.ts`: Update test to use the new endpoint to define styleProperties.
* `dot-page-api.service.ts`: Update endpoint for styleProperties definition.
* `PageResource.java`: Send user to `pageResourceHelper` specifically to the **saveContent** and **saveContentletStyles** methods.
* `pageResourceHelper.java`: Extract and reuse method for convert a folder asset container ID to its Container identifier

This PR fixes: #33699

This PR fixes: #33699

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java [hunk 1]
```diff
@@ -496,23 +497,14 @@ protected Contentlet copyContentlet(final CopyContentletForm copyContentletForm,
         final Contentlet copiedContentlet   = tuple2._1();
         final Contentlet originalContentlet = tuple2._2();
         final String htmlPage   = copyContentletForm.getPageId();
-        String container        = copyContentletForm.getContainerId();
+        final String container  = castToOriginalContainerId(copyContentletForm.getContainerId(), user, pageMode.respectAnonPerms);
         final String contentId  = copyContentletForm.getContentId();
```
Comment (fabrizzio-dotCMS) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java:501]: please double check if this change is breaking the original logic

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java [hunk 2]
```diff
@@ -169,7 +169,8 @@ public List<ContentView> saveContent(final String pageId,
 
             if (UtilMethods.isSet(contentIds)) {
                 for (final String contentletId : contentIds) {
-                    final MultiTree multiTree = new MultiTree().setContainer(containerEntry.getContainerId())
+                    final MultiTree multiTree = new MultiTree()
+                            .setContainer(castToOriginalContainerId(containerEntry.getContainerId(), user, false))
```
Comment (fabrizzio-dotCMS) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java:173]: here we're passing false
and down below we are using pageMode.respectAnonPerms
why?

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java [hunk 3]
```diff
@@ -796,12 +818,17 @@ private void updateContentletStyles(final String pageId, final String contentlet
...
      * @param variantId       The variant identifier
+     * @param user            The user performing the action
      * @return A unique string key combining all parameters
      */
     private String buildMultiTreeLookupKey(final String container, final String instanceId,
-            final String contentlet, final String personalization, final String variantId) {
+            final String contentlet, final String personalization, final String variantId, final User user) {
+
+        // Resolve folder asset container IDs to actual identifiers for consistent lookups
+        final String originalContainerID = castToOriginalContainerId(container, user, false);
```
Comment (ihoffmann-dot) [inline dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java:828]: Following Fabrizzio's question, we are using false here too. Maybe we should check?

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java (hunk 1 above)
Comment (dario-daza) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java:501]: I extract the logic from here because I needed to reuse this logic to get the original ContainerId, and I added the params accordingly, so at least this part of the code is not being affected.

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java (hunk 2 above)
Comment (dario-daza) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java:173]: I'm sending `false` because it didn't affected the result that I wanted to achieve (convert the containerId). I just left the `pageMode.respectAnonPerms` on the method above to avoid breaking the current logic.
```java
final String container  = castToOriginalContainerId(copyContentletForm.getContainerId(), user, pageMode.respectAnonPerms);
```

File: dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java (hunk 3 above)
Comment (dario-daza) [reply dotCMS/src/main/java/com/dotcms/rest/api/v1/page/PageResourceHelper.java:828]: It was intentional I am sending false for the reason that I comment above.

**Review by ihoffmann-dot — APPROVED** [review]

**Review by fabrizzio-dotCMS — APPROVED** [review]
